### ISSUE TITLE
Feature request: Subset and download data by rgt

### ISSUE THREAD

**Issue by barcheck** [icesat2py/icepyx#101]
For some processing, it would be nice to be able to download all repeats of a single or a few rgts, without having to download anything else.

**Comment by JessicaS11** [comment]
@asteiker @wallinb @nicholas-kotlinski @tsutterley Do you know if this is something that's possible within the NSIDC or CMR APIs? I know it's been a topic of discussion whether or not it could be implemented on the user (icepyx) end.

**Comment by fspaolo** [comment]
Wouldn't this be a matter of just filtering the list of files by selecting the RGT number from the file name? If the user specifies an area, `icepyx` could query every granule that intersects that area and select only the specific RGTs (and download just that area or the complete orbits).

**Comment by wallinb** [comment]
I think this is possible in the CMR/EGI queries too, though it might require using related orbit parameters instead of the RGT number directly, e.g. equator crossing longitude: [https://cmr.earthdata.nasa.gov/search/granules?short_name=ATL03&page_size=2000&page_num=1&equator_crossing_longitude=103.4,103.5](https://cmr.earthdata.nasa.gov/search/granules?short_name=ATL03&page_size=2000&page_num=1&equator_crossing_longitude=103.4,103.5)

**Comment by tsutterley** [comment]
I think we could also use the orbit_number parameter to map from RGT.  That way we don't have to download a larger zip file of unwanted RGTs or figure out the mapping between RGT and crossing longitude.
```
cmr_orbit_number = RGT + np.arange(ncycles)*1387 + 201
```

For example:  
- Track 314 Cycle 1: https://cmr.earthdata.nasa.gov/search/granules?short_name=ATL03&page_size=2000&page_num=1&orbit_number=515&version=003
- Track 314 Cycle 2: https://cmr.earthdata.nasa.gov/search/granules?short_name=ATL03&page_size=2000&page_num=1&orbit_number=1902&version=003
- Track 314 Cycle 3: https://cmr.earthdata.nasa.gov/search/granules?short_name=ATL03&page_size=2000&page_num=1&orbit_number=3289&version=003

**Comment by tsutterley** [comment]
What i put in the PR is kinda a combo of @fspaolo and @wallinb 's suggestions.  I parsed the filenames from `avail_granules(ids=True)` to get the list of RGTs and cycles that are available from a given spatial and temporal search domain.  I then added keywords to the query to be able to reduce to a given cycle or RGT using the `cmr_orbit_parameter` function listed above.  Gives you the option to search both in space and time using the existing spatial search queries and the new orbit search queries.
